src/infill.cpp

```diff
@@ -48,12 +52,127 @@ void Infill::generate(Polygons& result_polygons, Polygons& result_lines)
+    // with an odd number of lines, sides need to be swapped around
+    if(rtMod == 1){
+      rtMod = (lineCountX + int(rotation / 90)) % 2;
+    }
+
+    // draw non-horizontal walls of octohedrons
+    Polygons po;
+    PolygonRef p = po.newPoly();
+    for(unsigned int ly=0; ly < lineCountY;){
+      for(size_t it = 0; it < 2; ly++, it++){
+        int side = (2*((ly + it + rtMod) % 2) - 1);
+        int y = (ly * lineSpacing) + boundary.min.Y + lineSpacing / 2 - (offset/2 * side);
+        int x = boundary.min.X-(offset/2);
+        if(it == 1){
+          x = (lineCountX * (lineSpacing)) + boundary.min.X + lineSpacing / 2 - (offset/2);
+        }
+        p.add(Point(x,y));
+        for(unsigned int lx=0; lx < lineCountX; lx++){
+          if(it == 1){
+            side = (2*((lx + ly + it + rtMod + lineCountX) % 2) - 1);
+            y = (ly * lineSpacing) + boundary.min.Y + lineSpacing / 2 + (offset/2 * side);
+            x = ((lineCountX - lx - 1) * lineSpacing) + boundary.min.X + lineSpacing / 2;
+            p.add(Point(x+lineSpacing-abs(offset/2), y));
+            p.add(Point(x+abs(offset/2), y));
+          } else {
+            side = (2*((lx + ly + it + rtMod) % 2) - 1);
+            y = (ly * lineSpacing) + boundary.min.Y + lineSpacing / 2 + (offset/2 * side);
+            x = (lx * lineSpacing) + boundary.min.X + lineSpacing / 2;
+            p.add(Point(x+abs(offset/2), y));
+            p.add(Point(x+lineSpacing-abs(offset/2), y));
+          }
+        }
+        x = (lineCountX * lineSpacing) + boundary.min.X + lineSpacing / 2 - (offset/2);
+        if(it == 1){
+          x = boundary.min.X-(offset/2);
+        }
+        y = (ly * lineSpacing) + boundary.min.Y + lineSpacing / 2 - (offset/2 * side);
+        p.add(Point(x,y));
+      }
+    }
+    // Generate tops / bottoms of octohedrons
+    if(abs((abs(offset) - Zscale/4)) < (extrusionWidth/2)){
+      uint64_t startLine = (offset < 0) ? 0 : 1;
+      uint64_t coverWidth = OCTSLEN(lineSpacing);
+      std::vector<Point> points;
+      for(size_t xi = 0; xi < (lineCountX+1); xi++){
+        for(size_t yi = 0; yi < (lineCountY); yi += 2){
+          points.push_back(Point(boundary.min.X + OCTDLEN(lineSpacing)
+                                 + (xi - startLine + rtMod) * lineSpacing,
+                                 boundary.min.Y + OCTDLEN(lineSpacing)
+                                 + (yi + (xi%2)) * lineSpacing
+                                 + extrusionWidth/2));
+        }
+      }
+      uint64_t order = 0;
+      for(Point pp : points){
+        PolygonRef p = po.newPoly();
+        for(size_t yi = 0; yi <= coverWidth; yi += extrusionWidth) {
+          if(order == 0){
+            p.add(Point(pp.X, pp.Y + yi));
+            p.add(Point(pp.X + coverWidth + extrusionWidth, pp.Y + yi));
+          } else {
+            p.add(Point(pp.X + coverWidth + extrusionWidth, pp.Y + yi));
+            p.add(Point(pp.X, pp.Y + yi));
```

Comment:
You're adding horizontal lines here. Diagonal lines generally produce better results due to being able to accelerate on two axes.